src/components/HomeLandingPage/HomeLandingPage.tsx

```diff
@@ -40,24 +43,71 @@ const HomeLandingPage = () => {
   useAllEvents();
 
   // Fetch events from cache
-  const { data } = useQuery(GET_ALL_EVENTS);
+  const { data } = useQuery(FETCH_ALL_EVENTS);
+  const [editEvent] = useMutation(EDIT_EVENT);
+  const [deleteEvent] = useMutation(DELETE_EVENT, {
+    update(cache, { data: { deleteEvent } }) {
+      if (!deleteEvent) {
+        return;
+      }
+      let { events } = cache.readQuery<any>({
+        query: FETCH_ALL_EVENTS,
+      });
+
+      events = events.filter((event) => event.id !== selectedEvent!.id);
+
+      cache.writeQuery({
+        query: FETCH_ALL_EVENTS,
+        data: { events },
+      });
+    },
+  });
   const events: Array<Event> = data ? data.events : [];
 
+  const handleArchiveEvent = async (event: Event) => {
+    await editEvent({
```

Comment:
`await` statements for these handle- callbacks seem a little inelegant - what are they used for right now and can they be potentially refactored (e.g. with promise chaining)?

Reply:
Perhaps the use of await can be replaced by `onCompleted` and `onError` options on the mutation.

https://www.apollographql.com/docs/react/data/mutations/#options

Reply:
good call - those `await`'s were there back when I refetched events, but I can take those out now

Reply:
except for `deleteEvent`, which I refactored to using `useEffect`